API/Controllers/ProjectController.cs

```diff
@@ -252,41 +261,68 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
                 return NotFound(problem);
             }
 
-            mapper.Map(projectResource, project);
+            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
+            bool isAllowed = userService.UserHasScope(user.IdentityId, nameof(Defaults.Scopes.ProjectWrite));
+
+            if(!(project.UserId == user.Id || isAllowed))
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to edit the project.",
+                    Detail = "The user is not allowed to edit the project.",
+                    Instance = "2E765D18-8EBC-4117-8F9E-B800E8967038"
+                };
+                return Unauthorized(problem);
+            }
+
+            // Upload the new file if there is one
             File file = null;
             if(projectResource.FileId != 0)
             {
+                if(project.ProjectIconId != 0)
+                {
+                    File fileToDelete = await fileService.FindAsync(project.ProjectIconId.Value);
+                    // Remove the file from the filesystem
+                    fileUploader.DeleteFileFromDirectory(fileToDelete);
+                    // Remove file from DB
+                    await fileService.RemoveAsync(project.ProjectIconId.Value);
+                   
+                   
+                    fileService.Save();
+                }
+
+                // Get the uploaded file
                 file = await fileService.FindAsync(projectResource.FileId);
-                project.ProjectIcon = file;
-            }
 
-            if(projectResource.FileId != 0 && file == null)
-            {
-                ProblemDetails problem = new ProblemDetails
-                 {
-                     Title = "File was not found.",
-                     Detail = "The specified file was not found while updating project.",
-                     Instance = "69166D3D-6D34-4050-BD25-71F1BEBE43D3"
-                 };
-                return BadRequest(problem);
+                if(file != null)
+                {
+                    project.ProjectIcon = file;
+
+                } else
+                {
+                    ProblemDetails problem = new ProblemDetails
+                    {
+                        Title = "File was not found.",
+                        Detail = "The specified file was not found while updating project.",
+                        Instance = "69166D3D-6D34-4050-BD25-71F1BEBE43D3"
```

Comment:
GUID is not unique, generate a fresh one using Visual Studio or an online tool, please.